Subject: [PATCH 10/21] Hide the "show/hide overlays" when no overlay installed

diff --git a/src/com/android/settings/applications/ManageApplications.java b/src/com/android/settings/applications/ManageApplications.java
index ed617814b3..c3bc4d4a94 100644
--- a/src/com/android/settings/applications/ManageApplications.java
+++ b/src/com/android/settings/applications/ManageApplications.java
@@ -195,6 +195,10 @@ public class ManageApplications extends InstrumentedFragment
     private boolean mShowSubstratum;
     private boolean mShowSubstratumIcons;
 
+    // if app and icon overlay installed
+    private boolean mAppOverlayInstalled;
+    private boolean mIconOverlayInstalled;
+
     private ApplicationsState mApplicationsState;
 
     public int mListType;
@@ -556,6 +560,9 @@ public class ManageApplications extends InstrumentedFragment
     }
 
     void updateOptionsMenu() {
+        mAppOverlayInstalled = isOverlayInstalled("app");
+        mIconOverlayInstalled = isOverlayInstalled("icon");
+
         if (mOptionsMenu == null) {
             return;
         }
@@ -575,13 +582,13 @@ public class ManageApplications extends InstrumentedFragment
                 && mListType != LIST_TYPE_HIGH_POWER);
 
         mOptionsMenu.findItem(R.id.show_substratum).setVisible(!mShowSubstratum
-                && mListType != LIST_TYPE_HIGH_POWER);
+                && mListType != LIST_TYPE_HIGH_POWER && mAppOverlayInstalled);
         mOptionsMenu.findItem(R.id.hide_substratum).setVisible(mShowSubstratum
-                && mListType != LIST_TYPE_HIGH_POWER);
+                && mListType != LIST_TYPE_HIGH_POWER && mAppOverlayInstalled);
         mOptionsMenu.findItem(R.id.show_substratum_icons).setVisible(!mShowSubstratumIcons
-                && mListType != LIST_TYPE_HIGH_POWER);
+                && mListType != LIST_TYPE_HIGH_POWER && mIconOverlayInstalled);
         mOptionsMenu.findItem(R.id.hide_substratum_icons).setVisible(mShowSubstratumIcons
-                && mListType != LIST_TYPE_HIGH_POWER);
+                && mListType != LIST_TYPE_HIGH_POWER && mIconOverlayInstalled);
     }
 
     @Override
@@ -670,6 +677,29 @@ public class ManageApplications extends InstrumentedFragment
         mFilterAdapter.setFilterEnabled(FILTER_APPS_DISABLED, hasDisabledApps);
     }
 
+    boolean isOverlayInstalled(String type) {
+        List<ApplicationInfo> packages = getActivity().getPackageManager()
+                .getInstalledApplications(PackageManager.GET_META_DATA);
+
+        for (ApplicationInfo packageInfo : packages) {
+            if (packageInfo.metaData != null) {
+                if (type.equals("app")) {
+                    if (packageInfo.metaData
+                                    .getString("Substratum_Parent") != null) {
+                        return true;
+                    }
+                }
+                if (type.equals("icon")) {
+                    if (packageInfo.metaData
+                                    .getString("Substratum_IconPack") != null) {
+                        return true;
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
     static class FilterSpinnerAdapter extends ArrayAdapter<CharSequence> {
 
         private final ManageApplications mManageApplications;
