Subject: [PATCH 21/21] Settings: Support color engine

diff --git a/src/com/android/settings/PreviewPagerAdapter.java b/src/com/android/settings/PreviewPagerAdapter.java
index 8158c80b36..d802316402 100644
--- a/src/com/android/settings/PreviewPagerAdapter.java
+++ b/src/com/android/settings/PreviewPagerAdapter.java
@@ -74,7 +74,7 @@ public PreviewPagerAdapter(Context context, boolean isLayoutRtl,
                 // Create a new configuration for the specified value. It won't
                 // have any theme set, so manually apply the current theme.
                 final Context configContext = context.createConfigurationContext(configurations[j]);
-                configContext.setTheme(context.getThemeResId());
+                configContext.getTheme().setTo(context.getTheme());
 
                 final LayoutInflater configInflater = LayoutInflater.from(configContext);
                 final ViewStub sampleViewStub = new ViewStub(configContext);
diff --git a/res/values-land/themes.xml b/res/values-land/themes.xml
index d24bb39d61..6d8cc639ba 100644
--- a/res/values-land/themes.xml
+++ b/res/values-land/themes.xml
@@ -15,7 +15,7 @@
   ~ limitations under the License
   -->
 <resources>
-    <style name="Theme.ConfirmDeviceCredentials" parent="Theme_SubSettings">
+    <style name="Theme.ConfirmDeviceCredentials" parent="Theme.SubSettings">
         <item name="android:windowActionBar">false</item>
         <item name="android:windowNoTitle">true</item>
     </style>
diff --git a/res/values-sw600dp/themes.xml b/res/values-sw600dp/themes.xml
index e3dfa94758..0f685963b0 100644
--- a/res/values-sw600dp/themes.xml
+++ b/res/values-sw600dp/themes.xml
@@ -16,7 +16,7 @@
   ~ limitations under the License
   -->
 <resources>
-    <style name="Theme.ConfirmDeviceCredentials" parent="Theme_SubSettings">
+    <style name="Theme.ConfirmDeviceCredentials" parent="Theme.SubSettings">
       <item name="confirmDeviceCredentialsSideMargin">0dp</item>
       <item name="confirmDeviceCredentialsTopMargin">64dp</item>
     </style>
diff --git a/res/values/themes.xml b/res/values/themes.xml
index b4d2d1533d..8fbd894deb 100644
--- a/res/values/themes.xml
+++ b/res/values/themes.xml
@@ -168,7 +168,7 @@
         <item name="@*android:errorColor">@color/lock_pattern_view_error_color</item>
     </style>
 
-    <style name="Theme_SubSettings">
+    <style name="Theme.SubSettings" parent="Theme.Settings">
         <!-- Redefine the ActionBar style for contentInsetStart -->
         <item name="android:actionBarStyle">@style/Theme.ActionBar.SubSettings</item>
 
@@ -251,7 +251,7 @@
         <item name="preferenceBackgroundColor">@android:color/transparent</item>
     </style>
 
-    <style name="Theme.ConfirmDeviceCredentials" parent="Theme_SubSettings">
+    <style name="Theme.ConfirmDeviceCredentials" parent="Theme.SubSettings">
         <item name="confirmDeviceCredentialsSideMargin">16dp</item>
         <item name="confirmDeviceCredentialsTopMargin">16dp</item>
     </style>
diff --git a/src/com/android/settings/SettingsActivity.java b/src/com/android/settings/SettingsActivity.java
index b51f92ae54..38d4aacb5a 100644
--- a/src/com/android/settings/SettingsActivity.java
+++ b/src/com/android/settings/SettingsActivity.java
@@ -560,6 +560,29 @@ private static boolean isLikeShortCutIntent(final Intent intent) {
 
     @Override
     protected void onCreate(Bundle savedState) {
+
+        // Should happen before any call to getIntent()
+        getMetaData();
+        final Intent intent = getIntent();
+
+        // This is a "Sub Settings" when:
+        // - this is a real SubSettings
+        // - or :settings:show_fragment_as_subsetting is passed to the Intent
+        final boolean isSubSettings = this instanceof SubSettings ||
+                intent.getBooleanExtra(EXTRA_SHOW_FRAGMENT_AS_SUBSETTING, false);
+
+        boolean themeSet = false;
+        // If this is a sub settings, then apply the SubSettings Theme for the ActionBar content insets
+        if (isSubSettings) {
+            // Check also that we are not a Theme Dialog as we don't want to override them
+            final int themeResId = getThemeResId();
+            if (themeResId != R.style.Theme_DialogWhenLarge &&
+                    themeResId != R.style.Theme_SubSettingsDialogWhenLarge) {
+                setTheme(R.style.Theme_SubSettings);
+                themeSet = true;
+            }
+        }
+
         final int themeMode = Secure.getInt(getContentResolver(),
                 Secure.THEME_PRIMARY_COLOR, 0);
         final int accentColor = Secure.getInt(getContentResolver(),
@@ -568,8 +591,10 @@ protected void onCreate(Bundle savedState) {
         if (mThemeManager != null) {
             mThemeManager.addCallback(mThemeCallback);
         }
-        setTheme(R.style.Theme_Settings);
         if (themeMode != 0 || accentColor != 0) {
+            if (!themeSet) {
+                setTheme(R.style.Theme_Settings);
+            }
             getTheme().applyStyle(mTheme, true);
         }
         if (themeMode == 2) {
@@ -578,10 +603,6 @@ protected void onCreate(Bundle savedState) {
         super.onCreate(savedState);
         long startTime = System.currentTimeMillis();
 
-        // Should happen before any call to getIntent()
-        getMetaData();
-
-        final Intent intent = getIntent();
         if (intent.hasExtra(EXTRA_LAUNCH_ACTIVITY_ACTION)) {
             if (mActivityAction != null) {
                try{
@@ -619,22 +640,6 @@ protected void onCreate(Bundle savedState) {
                 || className.equals(Settings.PersonalSettings.class.getName())
                 || className.equals(Settings.WirelessSettings.class.getName());
 
-        // This is a "Sub Settings" when:
-        // - this is a real SubSettings
-        // - or :settings:show_fragment_as_subsetting is passed to the Intent
-        final boolean isSubSettings = this instanceof SubSettings ||
-                intent.getBooleanExtra(EXTRA_SHOW_FRAGMENT_AS_SUBSETTING, false);
-
-        // If this is a sub settings, then apply the SubSettings Theme for the ActionBar content insets
-        if (isSubSettings) {
-            // Check also that we are not a Theme Dialog as we don't want to override them
-            final int themeResId = getThemeResId();
-            if (themeResId != R.style.Theme_DialogWhenLarge &&
-                    themeResId != R.style.Theme_SubSettingsDialogWhenLarge) {
-                // Don't override theme to retain selected accent colors
-                getTheme().applyStyle(R.style.Theme_SubSettings, true);
-            }
-        }
 
         setContentView(mIsShowingDashboard ?
                 R.layout.settings_main_dashboard : R.layout.settings_main_prefs);
