Subject: [PATCH] Fix Settings FC after Color Engine merge

diff --git a/src/com/android/settings/ChooseLockGeneric.java b/src/com/android/settings/ChooseLockGeneric.java
index cd0688581e..35ee2907a9 100644
--- a/src/com/android/settings/ChooseLockGeneric.java
+++ b/src/com/android/settings/ChooseLockGeneric.java
@@ -248,15 +248,17 @@ protected void addHeaderView() {
         public void onViewCreated(View view, Bundle savedInstanceState) {
             super.onViewCreated(view, savedInstanceState);
             if (mForFingerprint) {
-                GlifPreferenceLayout layout = (GlifPreferenceLayout) view;
-                layout.setDividerItemDecoration(new SettingsDividerItemDecoration(getContext()));
+                if (view instanceof GlifPreferenceLayout) {
+                    GlifPreferenceLayout layout = (GlifPreferenceLayout) view;
+                    layout.setDividerItemDecoration(new SettingsDividerItemDecoration(getContext()));
 
-                layout.setIcon(getContext().getDrawable(R.drawable.ic_lock));
-                layout.setHeaderText(getActivity().getTitle());
+                    layout.setIcon(getContext().getDrawable(R.drawable.ic_lock));
+                    layout.setHeaderText(getActivity().getTitle());
 
-                // Use the dividers in SetupWizardRecyclerLayout. Suppress the dividers in
-                // PreferenceFragment.
-                setDivider(null);
+                    // Use the dividers in SetupWizardRecyclerLayout. Suppress the dividers in
+                    // PreferenceFragment.
+                    setDivider(null);
+                }
             }
         }
 
@@ -264,8 +266,10 @@ public void onViewCreated(View view, Bundle savedInstanceState) {
         public RecyclerView onCreateRecyclerView(LayoutInflater inflater, ViewGroup parent,
                 Bundle savedInstanceState) {
             if (mForFingerprint) {
-                GlifPreferenceLayout layout = (GlifPreferenceLayout) parent;
-                return layout.onCreateRecyclerView(inflater, parent, savedInstanceState);
+                if (parent instanceof GlifPreferenceLayout) {
+                    GlifPreferenceLayout layout = (GlifPreferenceLayout) parent;
+                    return layout.onCreateRecyclerView(inflater, parent, savedInstanceState);
+                }
             }
             return super.onCreateRecyclerView(inflater, parent, savedInstanceState);
         }
