 Samsung-STE: twrp fix offline charging

diff --git a/minui21/graphics.c b/minui21/graphics.c
index 580af29..08e6350 100644
--- a/minui21/graphics.c
+++ b/minui21/graphics.c
@@ -403,7 +403,7 @@
     gr_font->texture->format = GGL_PIXEL_FORMAT_A_8;
 }
 
-int gr_init(void)
+int gr_init_real(void)
 {
     gglInit(&gr_context);
     GGLContext *gl = gr_context;
@@ -450,6 +450,25 @@
     return 0;
 }
 
+/*
+ * FIXME: This is a total hack.
+ *
+ * Round 1
+ * framebuffer: fd 4 (480 x 800)
+ *
+ * Round 2
+ * framebuffer: fd 6 (480 x 800)
+ *
+*/
+int gr_init(void) {
+    int ret;
+    // Round 1
+    gr_init_real();
+    // Return the result of round 2
+    ret = gr_init_real();
+    return ret;
+}
+
 void gr_exit(void)
 {
     free_overlay(gr_fb_fd);
diff --git a/minui/graphics.cpp b/minui/graphics.cpp
index eceaeee..eb454c7 100644
--- a/minui/graphics.cpp
+++ b/minui/graphics.cpp
@@ -578,7 +578,7 @@
     gr_draw = gr_backend->flip(gr_backend);
 }
 
-int gr_init(void)
+int gr_init_real(void)
 {
     gr_init_font();
     gr_draw = NULL;
@@ -634,6 +634,25 @@
     return 0;
 }
 
+/*
+ * FIXME: This is a total hack.
+ *
+ * Round 1
+ * framebuffer: fd 4 (480 x 800)
+ *
+ * Round 2
+ * framebuffer: fd 6 (480 x 800)
+ *
+*/
+int gr_init(void) {
+    int ret;
+    // Round 1
+    gr_init_real();
+    // Return the result of round 2
+    ret = gr_init_real();
+    return ret;
+}
+
 void gr_exit(void)
 {
     gr_backend->exit(gr_backend);
