twrp: add /ramdisk to backup list 

diff --git a/data.cpp b/data.cpp
index 33a43d8..298c970 100644
--- a/data.cpp
+++ b/data.cpp
@@ -672,7 +672,7 @@ void DataManager::SetDefaultValues()
 #ifdef TW_HAS_NO_BOOT_PARTITION
 	mPersist.SetValue("tw_backup_list", "/system;/data;");
 #else
-	mPersist.SetValue("tw_backup_list", "/system;/data;/boot;");
+	mPersist.SetValue("tw_backup_list", "/system;/data;/boot;/ramdisk;");
 #endif
 	mConst.SetValue(TW_MIN_SYSTEM_VAR, TW_MIN_SYSTEM_SIZE);
 	mData.SetValue(TW_BACKUP_NAME, "(Auto Generate)");
diff --git a/openrecoveryscript.cpp b/openrecoveryscript.cpp
index 33a43d8..298c980 100644
--- a/openrecoveryscript.cpp
+++ b/openrecoveryscript.cpp
@@ -264,9 +264,9 @@ int OpenRecoveryScript::run_script_file(void) {
						} else if ((value2[i] == 'C' || value2[i] == 'c') && Partition_List.find("/cache;") != string::npos) {
							Restore_List += "/cache;";
							gui_msg("cache=Cache");
-						} else if ((value2[i] == 'R' || value2[i] == 'r') && Partition_List.find("/recovery;") != string::npos) {
-							Restore_List += "/recovery;";
-							gui_msg("recovery=Recovery");
+						} else if ((value2[i] == 'R' || value2[i] == 'r') && Partition_List.find("/ramdisk;") != string::npos) {
+							Restore_List += "/ramdisk;";
+							gui_msg("ramdisk=Ramdisk");
 						} else if ((value2[i] == 'B' || value2[i] == 'b') && Partition_List.find("/boot;") != string::npos) {
 							Restore_List += "/boot;";
 							gui_msg("boot=Boot");
diff --git a/partition.cpp b/partition.cpp
index 33a43d8..298c990 100644
--- a/partition.cpp
+++ b/partition.cpp
@@ -380,6 +380,11 @@ bool TWPartition::Process_Fstab_Line(string Line, bool Display_Error) {
 			Backup_Display_Name = Display_Name;
 			DataManager::SetValue("tw_boot_is_mountable", 1);
 			Can_Be_Backed_Up = true;
+		} else if (Mount_Point == "/ramdisk") {
+			Display_Name = "ramdisk";
+			Backup_Display_Name = Display_Name;
+			Wipe_Available_in_GUI = true;
+			Can_Be_Backed_Up = true;
 		} else if (Mount_Point == "/vendor") {
 			Display_Name = "Vendor";
 			Backup_Display_Name = Display_Name;
