Subject: [PATCH] build: no ninjas please

diff --git a/envsetup.sh b/envsetup.sh
index e392671..a6ab536 100644
--- a/envsetup.sh
+++ b/envsetup.sh
@@ -301,6 +301,7 @@
     # With this environment variable new GCC can apply colors to warnings/errors
     export GCC_COLORS='error=01;31:warning=01;35:note=01;36:caret=01;32:locus=01:quote=01'
     export ASAN_OPTIONS=detect_leaks=0
+    export USE_NINJA=false
 }
 
 function set_sequence_number()
