Subject: [PATCH] libstagefright: Revert Track graphic buffer mode in OMXNodeInstance 

diff --git a/media/libstagefright/include/OMXNodeInstance.h b/media/libstagefright/include/OMXNodeInstance.h
index 7a7bee7..f7ff0b7 100644
--- a/media/libstagefright/include/OMXNodeInstance.h
+++ b/media/libstagefright/include/OMXNodeInstance.h
@@ -175,7 +175,7 @@
     KeyedVector<OMX::buffer_id, OMX_BUFFERHEADERTYPE *> mBufferIDToBufferHeader;
     KeyedVector<OMX_BUFFERHEADERTYPE *, OMX::buffer_id> mBufferHeaderToBufferID;
 
-    // metadata and secure buffer types and graphic buffer mode tracking
+    // metadata and secure buffer type tracking
     MetadataBufferType mMetadataType[2];
     enum SecureBufferType {
         kSecureBufferTypeUnknown,
@@ -183,7 +183,6 @@
         kSecureBufferTypeNativeHandle,
     };
     SecureBufferType mSecureBufferType[2];
-    bool mGraphicBufferEnabled[2];
 
     // For debug support
     char *mName;
diff --git a/media/libstagefright/omx/OMXNodeInstance.cpp b/media/libstagefright/omx/OMXNodeInstance.cpp
index cf7f62d..f49c734 100644
--- a/media/libstagefright/omx/OMXNodeInstance.cpp
+++ b/media/libstagefright/omx/OMXNodeInstance.cpp
@@ -228,8 +228,6 @@
     mMetadataType[1] = kMetadataBufferTypeInvalid;
     mSecureBufferType[0] = kSecureBufferTypeUnknown;
     mSecureBufferType[1] = kSecureBufferTypeUnknown;
-    mGraphicBufferEnabled[0] = false;
-    mGraphicBufferEnabled[1] = false;
     mIsSecure = AString(name).endsWith(".secure");
     atomic_store(&mDying, false);
 }
@@ -568,12 +566,6 @@
             } else if (mSecureBufferType[portIndex] == kSecureBufferTypeUnknown) {
                 mSecureBufferType[portIndex] = kSecureBufferTypeOpaque;
             }
-        } else {
-            if (err == OMX_ErrorNone) {
-                mGraphicBufferEnabled[portIndex] = enable;
-            } else if (enable) {
-                mGraphicBufferEnabled[portIndex] = false;
-            }
         }
     } else {
         CLOG_ERROR_IF(enable, getExtensionIndex, err, "%s", name);
@@ -810,13 +802,6 @@
         return BAD_VALUE;
     }
 
-    if (mMetadataType[portIndex] == kMetadataBufferTypeInvalid
-            && mGraphicBufferEnabled[portIndex]) {
-        ALOGE("b/62948670");
-        android_errorWriteLog(0x534e4554, "62948670");
-        return INVALID_OPERATION;
-    }
-
     // metadata buffers are not connected cross process
     // use a backup buffer instead of the actual buffer
     BufferMeta *buffer_meta;
@@ -940,14 +925,6 @@
     }
     Mutex::Autolock autoLock(mLock);
 
-    if (!mGraphicBufferEnabled[portIndex]
-            || mMetadataType[portIndex] != kMetadataBufferTypeInvalid) {
-        // Report error if this is not in graphic buffer mode.
-        ALOGE("b/62948670");
-        android_errorWriteLog(0x534e4554, "62948670");
-        return INVALID_OPERATION;
-    }
-
     // See if the newer version of the extension is present.
     OMX_INDEXTYPE index;
 #ifndef STE_HARDWARE
