Subject: [PATCH] base: set scrolling to 0.006f

diff --git a/core/java/android/widget/AbsListView.java b/core/java/android/widget/AbsListView.java
index 811b7f41f069..9ca4310dceec 100644
--- a/core/java/android/widget/AbsListView.java
+++ b/core/java/android/widget/AbsListView.java
@@ -4630,6 +4630,7 @@ public void run() {
 
         FlingRunnable() {
             mScroller = new OverScroller(getContext());
+            mScroller.setFriction(0.006f);
         }
 
         void start(int initialVelocity) {
diff --git a/core/java/android/widget/ScrollView.java b/core/java/android/widget/ScrollView.java
index 0b854e3447f9..2010213bab80 100644
--- a/core/java/android/widget/ScrollView.java
+++ b/core/java/android/widget/ScrollView.java
@@ -241,6 +241,7 @@ public int getMaxScrollAmount() {
 
     private void initScrollView() {
         mScroller = new OverScroller(getContext());
+        mScroller.setFriction(0.006f);
         setFocusable(true);
         setDescendantFocusability(FOCUS_AFTER_DESCENDANTS);
         setWillNotDraw(false);
