 Add permissions to Browser package 

diff --git a/services/core/java/com/android/server/pm/DefaultPermissionGrantPolicy.java b/services/core/java/com/android/server/pm/DefaultPermissionGrantPolicy.java
index 4726ae7..28c18be 100644
--- a/services/core/java/com/android/server/pm/DefaultPermissionGrantPolicy.java
+++ b/services/core/java/com/android/server/pm/DefaultPermissionGrantPolicy.java
@@ -540,6 +540,8 @@ private void grantDefaultSystemHandlerPermissions(int userId) {
             if (browserPackage != null
                     && doesPackageSupportRuntimePermissions(browserPackage)) {
                 grantRuntimePermissionsLPw(browserPackage, LOCATION_PERMISSIONS, userId);
+                grantRuntimePermissionsLPw(browserPackage, CONTACTS_PERMISSIONS, userId);
+                grantRuntimePermissionsLPw(browserPackage, STORAGE_PERMISSIONS, userId);
             }
 
             // Voice interaction
@@ -914,6 +916,8 @@ public void grantDefaultPermissionsToDefaultBrowserLPr(String packageName, int u
         if (browserPackage != null
                 && doesPackageSupportRuntimePermissions(browserPackage)) {
             grantRuntimePermissionsLPw(browserPackage, LOCATION_PERMISSIONS, false, false, userId);
+            grantRuntimePermissionsLPw(browserPackage, CONTACTS_PERMISSIONS, false, false, userId);
+            grantRuntimePermissionsLPw(browserPackage, STORAGE_PERMISSIONS, false, false, userId);
         }
     }
 
