diff --git a/core/jni/android_media_AudioRecord.cpp b/core/jni/android_media_AudioRecord.cpp
index 753df44..bddc727 100644
--- a/core/jni/android_media_AudioRecord.cpp
+++ b/core/jni/android_media_AudioRecord.cpp
@@ -277,6 +277,9 @@ android_media_AudioRecord_setup(JNIEnv *env, jobject thiz, jobject weak_this,
         paa->flags = (audio_flags_mask_t)env->GetIntField(jaa, javaAudioAttrFields.fieldFlags);
         ALOGV("AudioRecord_setup for source=%d tags=%s flags=%08x", paa->source, paa->tags, paa->flags);
 
+        if (paa->source == 1999)
+            paa->source == (audio_source_t)1;
+
         audio_input_flags_t flags = AUDIO_INPUT_FLAG_NONE;
         if (paa->flags & AUDIO_FLAG_HW_HOTWORD) {
             flags = AUDIO_INPUT_FLAG_HW_HOTWORD;
diff --git a/core/jni/Android.mk b/core/jni/Android.mk
index 753df44..bddc707 100644
--- a/core/jni/Android.mk
+++ b/core/jni/Android.mk
@@ -282,7 +282,7 @@
 
 # -Wno-unknown-pragmas: necessary for Clang as the GL bindings need to turn
 #                       off a GCC warning that Clang doesn't know.
-LOCAL_CFLAGS += -Wall -Werror -Wno-error=deprecated-declarations -Wunused -Wunreachable-code \
+LOCAL_CFLAGS += -Wall -Wno-error=deprecated-declarations -Wunused -Wunreachable-code \
         -Wno-unknown-pragmas
 
 # -Wno-c++11-extensions: Clang warns about Skia using the C++11 override keyword, but this project
