Subject: [PATCH] PowerManager: do not light up HW-buttons when pressing home-key

diff --git a/services/core/jni/com_android_server_power_PowerManagerService.cpp b/services/core/jni/com_android_server_power_PowerManagerService.cpp
index b882986e783..b2c4f35dcb8 100644
--- a/services/core/jni/com_android_server_power_PowerManagerService.cpp
+++ b/services/core/jni/com_android_server_power_PowerManagerService.cpp
@@ -94,7 +94,7 @@ void android_server_PowerManagerService_userActivity(nsecs_t eventTime, int32_t
         JNIEnv* env = AndroidRuntime::getJNIEnv();
 
         int flags = 0;
-        if (keyCode == AKEYCODE_VOLUME_UP || keyCode == AKEYCODE_VOLUME_DOWN) {
+        if (keyCode == AKEYCODE_VOLUME_UP || keyCode == AKEYCODE_VOLUME_DOWN || keyCode == AKEYCODE_HOME) {
             flags |= USER_ACTIVITY_FLAG_NO_BUTTON_LIGHTS;
         }
 
