SHack: fix wifi autoconnection [2/3]

diff --git a/core/java/android/os/SystemClock.java b/core/java/android/os/SystemClock.java
index 00a5803..3b82593 100644
--- a/core/java/android/os/SystemClock.java
+++ b/core/java/android/os/SystemClock.java
@@ -172,13 +172,27 @@ public static boolean setCurrentTimeMillis(long millis) {
     native public static long elapsedRealtime();
 
     /**
+     * Returns milliseconds since boot, including time spent in sleep.
+     *
+     * @return elapsed milliseconds since boot.
+     */
+    native public static long elapsedRealtime1();
+
+    /**
      * Returns nanoseconds since boot, including time spent in sleep.
      *
      * @return elapsed nanoseconds since boot.
      */
     public static native long elapsedRealtimeNanos();
 
     /**
+     * Returns nanoseconds since boot, including time spent in sleep.
+     *
+     * @return elapsed nanoseconds since boot.
+     */
+    public static native long elapsedRealtimeNanos1();
+
+    /**
      * Returns milliseconds running in the current thread.
      * 
      * @return elapsed milliseconds in the thread
diff --git a/core/jni/android_os_SystemClock.cpp b/core/jni/android_os_SystemClock.cpp
index 00d6dc7..f56b619 100644
--- a/core/jni/android_os_SystemClock.cpp
+++ b/core/jni/android_os_SystemClock.cpp
@@ -55,6 +55,15 @@ static jlong android_os_SystemClock_elapsedRealtime(JNIEnv* env,
 }
 
 /*
+ * native public static long elapsedRealtime1();
+ */
+static jlong android_os_SystemClock_elapsedRealtime1(JNIEnv* env,
+        jobject clazz)
+{
+    return (jlong)elapsedRealtime1();
+}
+
+/*
  * native public static long currentThreadTimeMillis();
  */
 static jlong android_os_SystemClock_currentThreadTimeMillis(JNIEnv* env,
@@ -102,6 +111,15 @@ static jlong android_os_SystemClock_elapsedRealtimeNano(JNIEnv* env,
 }
 
 /*
+ * public static native long elapsedRealtimeNano1();
+ */
+static jlong android_os_SystemClock_elapsedRealtimeNano1(JNIEnv* env,
+        jobject clazz)
+{
+    return (jlong)elapsedRealtimeNano1();
+}
+
+/*
  * JNI registration.
  */
 static const JNINativeMethod gMethods[] = {
@@ -110,6 +128,8 @@ static const JNINativeMethod gMethods[] = {
             (void*) android_os_SystemClock_uptimeMillis },
     { "elapsedRealtime",      "()J",
             (void*) android_os_SystemClock_elapsedRealtime },
+    { "elapsedRealtime1",      "()J",
+            (void*) android_os_SystemClock_elapsedRealtime1 },
     { "currentThreadTimeMillis",      "()J",
             (void*) android_os_SystemClock_currentThreadTimeMillis },
     { "currentThreadTimeMicro",       "()J",
@@ -118,6 +138,8 @@ static const JNINativeMethod gMethods[] = {
             (void*) android_os_SystemClock_currentTimeMicro },
     { "elapsedRealtimeNanos",      "()J",
             (void*) android_os_SystemClock_elapsedRealtimeNano },
+    { "elapsedRealtimeNanos1",      "()J",
+            (void*) android_os_SystemClock_elapsedRealtimeNano1 },
 };
 int register_android_os_SystemClock(JNIEnv* env)
 {
