Subject: [PATCH] ParcelFileDescriptor: can we stop the panic here?

diff --git a/core/java/android/os/ParcelFileDescriptor.java b/core/java/android/os/ParcelFileDescriptor.java
index c091420a31d4..b3c7ced0c61b 100644
--- a/core/java/android/os/ParcelFileDescriptor.java
+++ b/core/java/android/os/ParcelFileDescriptor.java
@@ -178,7 +178,11 @@ public ParcelFileDescriptor(FileDescriptor fd, FileDescriptor commChannel) {
         mWrapped = null;
         mFd = fd;
         mCommFd = commChannel;
-        mGuard.open("close");
+        try {
+            mGuard.open("close");
+        } catch(Throwable e) {
+            Log.w("ParcelFileDescriptor", "Explicit termination method 'close' not called");
+        }
     }
 
     /**
