binder: protected against null Parcelable 

diff --git a/libs/binder/Parcel.cpp b/libs/binder/Parcel.cpp
index 4e122d0..064e119 100644
--- a/libs/binder/Parcel.cpp
+++ b/libs/binder/Parcel.cpp
@@ -2028,6 +2028,9 @@ wp<IBinder> Parcel::readWeakBinder() const
 }
 
 status_t Parcel::readParcelable(Parcelable* parcelable) const {
+    if (parcelable == nullptr) {
+        return BAD_VALUE;
+    }
     int32_t have_parcelable = 0;
     status_t status = readInt32(&have_parcelable);
     if (status != OK) {
