Subject: [PATCH] Disable dashboard suggestions [1/2]

diff --git a/core/java/android/provider/Settings.java b/core/java/android/provider/Settings.java
index c1c1509d771..c4c5e742bc6 100644
--- a/core/java/android/provider/Settings.java
+++ b/core/java/android/provider/Settings.java
@@ -3755,6 +3755,12 @@ public boolean validate(String value) {
         public static final String ENABLE_CONDITIONS = "enable_conditions";
 
         /**
+         * Disable dashboard suggestions in settings
+         * @hide
+         */
+        public static final String DISABLE_SUGGESTIONS = "disable_suggestions";
+
+        /**
          * Settings to backup. This is here so that it's in the same place as the settings
          * keys and easy to update.
          *
