 Toast icon show hide

diff --git a/core/java/android/widget/Toast.java b/core/java/android/widget/Toast.java
index 06e667c..9e3ac99 100644
--- a/core/java/android/widget/Toast.java
+++ b/core/java/android/widget/Toast.java
@@ -32,6 +32,7 @@
 import android.os.Message;
 import android.os.RemoteException;
 import android.os.ServiceManager;
+import android.os.SystemProperties;
 import android.util.Log;
 import android.view.Gravity;
 import android.view.LayoutInflater;
@@ -332,6 +333,9 @@
         };
 
         private final WindowManager.LayoutParams mParams = new WindowManager.LayoutParams();
+
+        private static final boolean mToasticonSwitch =
+                SystemProperties.getBoolean("ro.toast.icon_show", true);
 
         private static final int SHOW = 0;
         private static final int HIDE = 1;
@@ -436,18 +440,20 @@ public void handleShow() {
                 }
 
                 ImageView appIcon = (ImageView) mView.findViewById(android.R.id.icon);
-                if (appIcon != null) {
-                    ActivityManager am =
-                            (ActivityManager) context.getSystemService(Context.ACTIVITY_SERVICE);
-                    if (!am.isPackageInForeground(packageName)) {
-                        PackageManager pm = context.getPackageManager();
-                        Drawable icon = null;
-                        try {
-                            icon = pm.getApplicationIcon(packageName);
-                        } catch (PackageManager.NameNotFoundException e) {
-                            // nothing to do
-                        }
-                        appIcon.setImageDrawable(icon);
+                if (mToasticonSwitch) {
+                    if (appIcon != null) {
+                        ActivityManager am =
+                                (ActivityManager) context.getSystemService(Context.ACTIVITY_SERVICE);
+                        if (!am.isPackageInForeground(packageName)) {
+                            PackageManager pm = context.getPackageManager();
+                            Drawable icon = null;
+                            try {
+                                icon = pm.getApplicationIcon(packageName);
+                            } catch (PackageManager.NameNotFoundException e) {
+                                // nothing to do
+                            }
+                            appIcon.setImageDrawable(icon);
+                        }
                     }
                 }
                 mWM = (WindowManager)context.getSystemService(Context.WINDOW_SERVICE);
