Handle NPE due to threading race

diff --git a/media/java/android/media/MediaPlayer.java b/media/java/android/media/MediaPlayer.java
index 3687512..3660f61 100644
--- a/media/java/android/media/MediaPlayer.java
+++ b/media/java/android/media/MediaPlayer.java
@@ -2855,6 +2855,7 @@ public void handleMessage(Message msg) {
                 Log.w(TAG, "mediaplayer went away with unhandled events");
                 return;
             }
+            try {
             switch(msg.what) {
             case MEDIA_PREPARED:
                 try {
@@ -3033,6 +3034,13 @@ public void handleMessage(Message msg) {
                 Log.e(TAG, "Unknown message type " + msg.what);
                 return;
             }
+            } catch (NullPointerException e) {
+                /**
+                 * We may get an NPE even with the null checks above due
+                 * to threading issues.  Just ignore it.
+                 */
+                Log.e(TAG, "Unhandled NPE from message type " + msg.what);
+            }
         }
     }
 
