Subject: [PATCH] [BUG] java.lang.NullPointerException: name is null

diff --git a/core/java/android/view/AccessibilityInteractionController.java b/core/java/android/view/AccessibilityInteractionController.java
index 1c458ab..7f8b96d 100644
--- a/core/java/android/view/AccessibilityInteractionController.java
+++ b/core/java/android/view/AccessibilityInteractionController.java
@@ -236,7 +236,7 @@
             } else {
                 root = mViewRootImpl.mView;
             }
-            if (root != null) {
+            if (root != null && viewId != null) {
                 final int resolvedViewId = root.getContext().getResources()
                         .getIdentifier(viewId, null, null);
                 if (resolvedViewId <= 0) {
