Subject: [PATCH] Fix BatteryService

diff --git a/services/core/java/com/android/server/BatteryService.java b/services/core/java/com/android/server/BatteryService.java
index f5f7700..35ad416 100644
--- a/services/core/java/com/android/server/BatteryService.java
+++ b/services/core/java/com/android/server/BatteryService.java
@@ -211,7 +211,7 @@
         mWeakChgCutoffVoltageMv = SystemProperties.getInt("ro.cutoff_voltage_mv", 0);
          /* 2700mV UVLO voltage */
         if (mWeakChgCutoffVoltageMv > 2700)
-           mVoltageNowFile = new File("/sys/class/power_supply/battery/voltage_now");
+           mVoltageNowFile = new File("/sys/class/power_supply/ab8500_fg/voltage_now");
 
         mCriticalBatteryLevel = mContext.getResources().getInteger(
                 com.android.internal.R.integer.config_criticalBatteryWarningLevel);
