Subject: [PATCH] Fix pms systemReady NullpointerException

diff --git a/services/core/java/com/android/server/pm/DefaultPermissionGrantPolicy.java b/services/core/java/com/android/server/pm/DefaultPermissionGrantPolicy.java
index 5016ec0..95c4fbc 100644
--- a/services/core/java/com/android/server/pm/DefaultPermissionGrantPolicy.java
+++ b/services/core/java/com/android/server/pm/DefaultPermissionGrantPolicy.java
@@ -1027,7 +1027,7 @@
         if (!isDefaultPhoneOrSms && pkg.isUpdatedSystemApp()) {
             PackageSetting sysPs = mService.mSettings.getDisabledSystemPkgLPr(pkg.packageName);
             if (sysPs != null) {
-                if (sysPs.pkg.requestedPermissions.isEmpty()) {
+                if (sysPs.pkg == null || sysPs.pkg.requestedPermissions.isEmpty()) {
                     return;
                 }
                 if (!requestedPermissions.equals(sysPs.pkg.requestedPermissions)) {
