Subject: [PATCH] Fix UnsupportedOperationException: removeView(View) is not supported in AdapterView

diff --git a/core/java/android/widget/AdapterView.java b/core/java/android/widget/AdapterView.java
index fe6d91a..5691740 100644
--- a/core/java/android/widget/AdapterView.java
+++ b/core/java/android/widget/AdapterView.java
@@ -520,7 +520,9 @@
      */
     @Override
     public void removeView(View child) {
-        throw new UnsupportedOperationException("removeView(View) is not supported in AdapterView");
+        if (child == null) {
+            throw new UnsupportedOperationException("removeView(View) is not supported in AdapterView");
+        }
     }
 
     /**
