Subject: [PATCH] RecentTasks: Check task for null and return if yes

diff --git a/services/core/java/com/android/server/am/RecentTasks.java b/services/core/java/com/android/server/am/RecentTasks.java
index a6ebac44062..28f85c0d20b 100644
--- a/services/core/java/com/android/server/am/RecentTasks.java
+++ b/services/core/java/com/android/server/am/RecentTasks.java
@@ -512,6 +512,7 @@ private final boolean moveAffiliatedTasksToFront(TaskRecord task, int taskIndex)
     }
 
     final void addLocked(TaskRecord task) {
+        if (task == null) return;
         final boolean isAffiliated = task.mAffiliatedTaskId != task.taskId
                 || task.mNextAffiliateTaskId != INVALID_TASK_ID
                 || task.mPrevAffiliateTaskId != INVALID_TASK_ID;
