Subject: [PATCH] Fix issue FD leaking when processing mediascanner

diff --git a/media/jni/android_media_ExifInterface.cpp b/media/jni/android_media_ExifInterface.cpp
old mode 100644
new mode 100755
index 731deae..24ebec8
--- a/media/jni/android_media_ExifInterface.cpp
+++ b/media/jni/android_media_ExifInterface.cpp
@@ -379,11 +379,13 @@
 
     if (fgets(jpegSignature.get(), kJpegSignatureSize, file) == NULL) {
         ALOGI("Corrupted image.");
+        fclose(file);
         return NULL;
     }
 
     if (memcmp(jpegSignature.get(), kJpegSignatureChars, kJpegSignatureSize) == 0) {
         ALOGI("Should be a JPEG stream.");
+        fclose(file);
         return NULL;
     }
 
