Subject: [PATCH] Replaced duplicate lookup with unused assignment value.
Subject: [PATCH] Replaced duplicate lookup with unused assignment value in findActionPositionById.

diff --git a/v17/leanback/src/android/support/v17/leanback/app/GuidedStepFragment.java b/v17/leanback/src/android/support/v17/leanback/app/GuidedStepFragment.java
index 03fbe391f..73eacdf16 100644
--- a/v17/leanback/src/android/support/v17/leanback/app/GuidedStepFragment.java
+++ b/v17/leanback/src/android/support/v17/leanback/app/GuidedStepFragment.java
@@ -682,7 +682,7 @@ public int findButtonActionPositionById(long id) {
         if (mButtonActions != null) {
             for (int i = 0; i < mButtonActions.size(); i++) {
                 GuidedAction action = mButtonActions.get(i);
-                if (mButtonActions.get(i).getId() == id) {
+                if (action.getId() == id) {
                     return i;
                 }
             }
@@ -775,7 +775,7 @@ public int findActionPositionById(long id) {
         if (mActions != null) {
             for (int i = 0; i < mActions.size(); i++) {
                 GuidedAction action = mActions.get(i);
-                if (mActions.get(i).getId() == id) {
+                if (action.getId() == id) {
                     return i;
                 }
             }
