Subject: [PATCH] Fix SecurityException: MODE_WORLD_READABLE no longer supported

diff --git a/core/java/android/app/ContextImpl.java b/core/java/android/app/ContextImpl.java
index 006d91a..5691740 100644
--- a/core/java/android/app/ContextImpl.java
+++ b/core/java/android/app/ContextImpl.java
@@ -2131,10 +2131,14 @@
     private void checkMode(int mode) {
         if (getApplicationInfo().targetSdkVersion >= Build.VERSION_CODES.N) {
             if ((mode & MODE_WORLD_READABLE) != 0) {
-                throw new SecurityException("MODE_WORLD_READABLE no longer supported");
+                if (mode == 0) {
+                    throw new SecurityException("MODE_WORLD_READABLE no longer supported");
+                }
             }
             if ((mode & MODE_WORLD_WRITEABLE) != 0) {
-                throw new SecurityException("MODE_WORLD_WRITEABLE no longer supported");
+                if (mode == 0) {
+                    throw new SecurityException("MODE_WORLD_WRITEABLE no longer supported");
+                }
             }
         }
     }
