diff --git a/media/libstagefright/Android.mk b/media/libstagefright/Android.mk
index fb87de0..fbc4048 100644
--- a/media/libstagefright/Android.mk
+++ b/media/libstagefright/Android.mk
@@ -165,7 +165,7 @@ LOCAL_SHARED_LIBRARIES += \
 LOCAL_CFLAGS += -DUSE_LEGACY_RESCALING
 endif
 
-LOCAL_CFLAGS += -Wno-multichar -Werror -Wno-error=deprecated-declarations -Wall
+LOCAL_CFLAGS += -Wno-multichar -Wno-error=deprecated-declarations -Wall
 
 LOCAL_C_INCLUDES += $(call project-path-for,qcom-media)/mm-core/inc
 
diff --git a/media/libmediaplayerservice/Android.mk b/media/libmediaplayerservice/Android.mk
index fb87de0..fbc4049 100644
--- a/media/libmediaplayerservice/Android.mk
+++ b/media/libmediaplayerservice/Android.mk
@@ -73,7 +73,8 @@
     $(TOP)/frameworks/av/media/libavextensions                      \
     $(TOP)/frameworks/av/media/libstagefright/mpeg2ts               \
 
-LOCAL_CFLAGS += -Werror -Wno-error=deprecated-declarations -Wall
+LOCAL_CFLAGS += -Wno-error=deprecated-declarations -Wall
+
 # DOLBY_START
 ifeq ($(strip $(DOLBY_ENABLE)),true)
     LOCAL_CFLAGS += $(dolby_cflags)
diff --git a/media/libstagefright/id3/Android.mk b/media/libstagefright/id3/Android.mk
index fb87de0..0bc4049 100644
--- a/media/libstagefright/id3/Android.mk
+++ b/media/libstagefright/id3/Android.mk
@@ -19,7 +19,7 @@ include $(CLEAR_VARS)
 LOCAL_SRC_FILES := \
 	testid3.cpp
 
-LOCAL_CFLAGS += -Werror -Wall
+LOCAL_CFLAGS += -Wall
 LOCAL_CLANG := true
 
 LOCAL_SHARED_LIBRARIES := \
diff --git a/media/libavextensions/Android.mk b/media/libavextensions/Android.mk
index fb87de0..fbc4008 100644
--- a/media/libavextensions/Android.mk
+++ b/media/libavextensions/Android.mk
@@ -16,7 +16,7 @@
         $(TOP)/frameworks/av/media/libstagefright \
         $(TOP)/frameworks/av/media/libstagefright/mpeg2ts \
 
-LOCAL_CFLAGS += -Wno-multichar -Werror
+LOCAL_CFLAGS += -Wno-multichar
 
 ifeq ($(TARGET_ENABLE_QC_AV_ENHANCEMENTS),true)
        LOCAL_CFLAGS += -DENABLE_AV_ENHANCEMENTS
diff --git a/services/audiopolicy/Android.mk b/services/audiopolicy/Android.mk
index fb87de0..fb14008 100644
--- a/services/audiopolicy/Android.mk
+++ b/services/audiopolicy/Android.mk
@@ -68,7 +68,7 @@ LOCAL_MULTILIB := $(AUDIOSERVER_MULTILIB)
     LOCAL_CFLAGS += $(dolby_cflags)
 endif
 # DOLBY_END
-LOCAL_CFLAGS += -Wall -Werror
+LOCAL_CFLAGS += -Wall
 
 include $(BUILD_SHARED_LIBRARY)
 
@@ -126,7 +126,7 @@ LOCAL_SHARED_LIBRARIES += libicuuc
 endif
 # DOLBY_END
 
-LOCAL_CFLAGS += -Wall -Werror
+LOCAL_CFLAGS += -Wall
 
 LOCAL_MULTILIB := $(AUDIOSERVER_MULTILIB)
 
@@ -156,7 +156,7 @@ LOCAL_C_INCLUDES += \
 endif
 # DOLBY_END
 
-LOCAL_CFLAGS := -Wall -Werror
+LOCAL_CFLAGS := -Wall
 
 LOCAL_MULTILIB := $(AUDIOSERVER_MULTILIB)
 
