Subject: [PATCH] Adjustments for M-compatible sensors ABI

diff --git a/core/jni/android_hardware_SensorManager.cpp b/core/jni/android_hardware_SensorManager.cpp
index 50f23bf..f29bec8 100644
--- a/core/jni/android_hardware_SensorManager.cpp
+++ b/core/jni/android_hardware_SensorManager.cpp
@@ -212,7 +212,11 @@
     SensorManager* mgr = reinterpret_cast<SensorManager*>(sensorManager);
 
     Sensor const* const* sensorList;
+#ifdef COMPAT_SENSORS_M
+    const ssize_t count = mgr->getSensorList(&sensorList);
+#else
     ssize_t count = mgr->getSensorList(&sensorList);
+#endif
     if (ssize_t(index) >= count) {
         return false;
     }
