Subject: [PATCH] Revert PLMN Selection Info to SIM Records
 Fix SIM connectivity

diff --git a/src/java/com/android/internal/telephony/uicc/SIMRecords.java b/src/java/com/android/internal/telephony/uicc/SIMRecords.java
index c97308a..bf9370c 100644
--- a/src/java/com/android/internal/telephony/uicc/SIMRecords.java
+++ b/src/java/com/android/internal/telephony/uicc/SIMRecords.java
@@ -171,11 +171,6 @@
     private static final int EVENT_GET_CSP_CPHS_DONE = 33 + SIM_RECORD_EVENT_BASE;
     private static final int EVENT_GET_GID1_DONE = 34 + SIM_RECORD_EVENT_BASE;
     private static final int EVENT_GET_GID2_DONE = 36 + SIM_RECORD_EVENT_BASE;
-    private static final int EVENT_GET_PLMN_W_ACT_DONE = 37 + SIM_RECORD_EVENT_BASE;
-    private static final int EVENT_GET_OPLMN_W_ACT_DONE = 38 + SIM_RECORD_EVENT_BASE;
-    private static final int EVENT_GET_HPLMN_W_ACT_DONE = 39 + SIM_RECORD_EVENT_BASE;
-    private static final int EVENT_GET_EHPLMN_DONE = 40 + SIM_RECORD_EVENT_BASE;
-    private static final int EVENT_GET_FPLMN_DONE = 41 + SIM_RECORD_EVENT_BASE;
 
     // TODO: Possibly move these to IccRecords.java
     private static final int SYSTEM_EVENT_BASE = 0x100;
@@ -275,11 +270,6 @@
         mPnnHomeName = null;
         mGid1 = null;
         mGid2 = null;
-        mPlmnActRecords = null;
-        mOplmnActRecords = null;
-        mHplmnActRecords = null;
-        mFplmns = null;
-        mEhplmns = null;
 
         mAdnCache.reset();
 
@@ -1250,72 +1262,6 @@
 
                 break;
 
-            case EVENT_GET_PLMN_W_ACT_DONE:
-                ar = (AsyncResult) msg.obj;
-                data = (byte[]) ar.result;
-
-                if (ar.exception != null || data == null) {
-                    loge("Failed getting User PLMN with Access Tech Records: " + ar.exception);
-                    break;
-                } else {
-                    log("Received a PlmnActRecord, raw=" + IccUtils.bytesToHexString(data));
-                    mPlmnActRecords = PlmnActRecord.getRecords(data);
-                    if (VDBG) log("PlmnActRecords=" + Arrays.toString(mPlmnActRecords));
-                }
-                break;
-
-            case EVENT_GET_OPLMN_W_ACT_DONE:
-                ar = (AsyncResult) msg.obj;
-                data = (byte[]) ar.result;
-
-                if (ar.exception != null || data == null) {
-                    loge("Failed getting Operator PLMN with Access Tech Records: "
-                            + ar.exception);
-                    break;
-                } else {
-                    log("Received a PlmnActRecord, raw=" + IccUtils.bytesToHexString(data));
-                    mOplmnActRecords = PlmnActRecord.getRecords(data);
-                    if (VDBG) log("OplmnActRecord[]=" + Arrays.toString(mOplmnActRecords));
-                }
-                break;
-
-            case EVENT_GET_HPLMN_W_ACT_DONE:
-                ar = (AsyncResult) msg.obj;
-                data = (byte[]) ar.result;
-
-                if (ar.exception != null || data == null) {
-                    loge("Failed getting Home PLMN with Access Tech Records: " + ar.exception);
-                    break;
-                } else {
-                    log("Received a PlmnActRecord, raw=" + IccUtils.bytesToHexString(data));
-                    mHplmnActRecords = PlmnActRecord.getRecords(data);
-                    log("HplmnActRecord[]=" + Arrays.toString(mHplmnActRecords));
-                }
-                break;
-
-            case EVENT_GET_EHPLMN_DONE:
-                ar = (AsyncResult) msg.obj;
-                data = (byte[]) ar.result;
-                if (ar.exception != null || data == null) {
-                    loge("Failed getting Equivalent Home PLMNs: " + ar.exception);
-                    break;
-                } else {
-                    mEhplmns = parseBcdPlmnList(data, "Equivalent Home");
-                }
-                break;
-
-            case EVENT_GET_FPLMN_DONE:
-                isRecordLoadResponse = true;
-                ar = (AsyncResult) msg.obj;
-                data = (byte[]) ar.result;
-                if (ar.exception != null || data == null) {
-                    loge("Failed getting Forbidden PLMNs: " + ar.exception);
-                    break;
-                } else {
-                    mFplmns = parseBcdPlmnList(data, "Forbidden");
-                }
-                break;
-
             case EVENT_CARRIER_CONFIG_CHANGED:
                 handleCarrierNameOverride();
                 break;
@@ -1706,17 +1630,6 @@
         mFh.loadEFTransparent(EF_GID2, obtainMessage(EVENT_GET_GID2_DONE));
         mRecordsToLoad++;
 
-        mFh.loadEFTransparent(EF_PLMN_W_ACT, obtainMessage(EVENT_GET_PLMN_W_ACT_DONE));
-
-        mFh.loadEFTransparent(EF_OPLMN_W_ACT, obtainMessage(EVENT_GET_OPLMN_W_ACT_DONE));
-
-        mFh.loadEFTransparent(EF_HPLMN_W_ACT, obtainMessage(EVENT_GET_HPLMN_W_ACT_DONE));
-
-        mFh.loadEFTransparent(EF_EHPLMN, obtainMessage(EVENT_GET_EHPLMN_DONE));
-
-        mFh.loadEFTransparent(EF_FPLMN, obtainMessage(EVENT_GET_FPLMN_DONE));
-        mRecordsToLoad++;
-
         loadEfLiAndEfPl();
         mFh.getEFLinearRecordSize(EF_SMS, obtainMessage(EVENT_GET_SMS_RECORD_SIZE_DONE));
 
@@ -2125,11 +2038,6 @@
         pw.println(" mUsimServiceTable=" + mUsimServiceTable);
         pw.println(" mGid1=" + mGid1);
         pw.println(" mGid2=" + mGid2);
-        pw.println(" mPlmnActRecords[]=" + Arrays.toString(mPlmnActRecords));
-        pw.println(" mOplmnActRecords[]=" + Arrays.toString(mOplmnActRecords));
-        pw.println(" mHplmnActRecords[]=" + Arrays.toString(mHplmnActRecords));
-        pw.println(" mFplmns[]=" + Arrays.toString(mFplmns));
-        pw.println(" mEhplmns[]=" + Arrays.toString(mEhplmns));
         pw.flush();
     }
 }
diff --git a/src/java/com/android/internal/telephony/uicc/UsimFileHandler.java b/src/java/com/android/internal/telephony/uicc/UsimFileHandler.java
index 066052f..b120eb6 100755
--- a/src/java/com/android/internal/telephony/uicc/UsimFileHandler.java
+++ b/src/java/com/android/internal/telephony/uicc/UsimFileHandler.java
@@ -61,13 +61,6 @@
         case EF_GID1:
         case EF_GID2:
         case EF_LI:
-        case EF_PLMN_W_ACT:
-        case EF_OPLMN_W_ACT:
-        case EF_HPLMN_W_ACT:
-        case EF_EHPLMN:
-        case EF_FPLMN:
-        case EF_LRPLMNSI:
-        case EF_HPPLMN:
             return MF_SIM + DF_ADF;
 
         case EF_PBR:
