Subject: [PATCH] Fix binderDied NetworkRequest

diff --git a/services/core/java/com/android/server/ConnectivityService.java b/services/core/java/com/android/server/ConnectivityService.java
index e184288..496c5c7 100644
--- a/services/core/java/com/android/server/ConnectivityService.java
+++ b/services/core/java/com/android/server/ConnectivityService.java
@@ -4134,11 +4134,13 @@
             mPendingIntent = null;
             enforceRequestCountLimit();
 
-            try {
-                mBinder.linkToDeath(this, 0);
-            } catch (RemoteException e) {
-                binderDied();
-            }
+            if (mBinder != null) {
+               try {
+                   mBinder.linkToDeath(this, 0);
+               } catch (RemoteException e) {
+                   binderDied();
+               }
+            }
         }
 
         private void enforceRequestCountLimit() {
