diff --git a/services/core/java/com/android/server/am/ActivityManagerService.java b/services/core/java/com/android/server/am/ActivityManagerService.java
index 1902999..015f6bb 100644
--- a/services/core/java/com/android/server/am/ActivityManagerService.java
+++ b/services/core/java/com/android/server/am/ActivityManagerService.java
@@ -9363,7 +9363,7 @@ private boolean isGetTasksAllowed(String caller, int callingPid, int callingUid)
         }
         if (!allowed) {
             if (DEBUG_TASKS) Slog.w(TAG, caller + ": caller " + callingUid
-                    + " does not hold REAL_GET_TASKS; limiting output");
+                    + " does not hold GET_TASKS; limiting output");
         }
         return allowed;
     }
@@ -14220,23 +14220,16 @@ private void fillInProcMemInfo(ProcessRecord app,
     @Override
     public List<ActivityManager.RunningAppProcessInfo> getRunningAppProcesses() {
         enforceNotIsolatedCaller("getRunningAppProcesses");
-
-        final int callingUid = Binder.getCallingUid();
-
         // Lazy instantiation of list
         List<ActivityManager.RunningAppProcessInfo> runList = null;
         final boolean allUsers = ActivityManager.checkUidPermission(INTERACT_ACROSS_USERS_FULL,
-                callingUid) == PackageManager.PERMISSION_GRANTED;
-        final int userId = UserHandle.getUserId(callingUid);
-        final boolean allUids = isGetTasksAllowed(
-                "getRunningAppProcesses", Binder.getCallingPid(), callingUid);
-
+                Binder.getCallingUid()) == PackageManager.PERMISSION_GRANTED;
+        int userId = UserHandle.getUserId(Binder.getCallingUid());
         synchronized (this) {
             // Iterate across all processes
-            for (int i = mLruProcesses.size() - 1; i >= 0; i--) {
+            for (int i = mLruProcesses.size() - 1; i >= 0; i--) {
                 ProcessRecord app = mLruProcesses.get(i);
-                if ((!allUsers && app.userId != userId)
-                        || (!allUids && app.uid != callingUid)) {
+                if (!allUsers && app.userId != userId) {
                     continue;
                 }
                 if ((app.thread != null) && (!app.crashing && !app.notResponding)) {
@@ -14260,7 +14253,7 @@ private void fillInProcMemInfo(ProcessRecord app,
                     //Slog.v(TAG, "Proc " + app.processName + ": imp=" + currApp.importance
                     //        + " lru=" + currApp.lru);
                     if (runList == null) {
-                        runList = new ArrayList<>();
+                        runList = new ArrayList<ActivityManager.RunningAppProcessInfo>();
                     }
                     runList.add(currApp);
                 }
