InputWrapper: opt out early if session == null 

diff --git a/core/java/android/inputmethodservice/IInputMethodWrapper.java b/core/java/android/inputmethodservice/IInputMethodWrapper.java
index fb06f01..7b23271 100644
--- a/core/java/android/inputmethodservice/IInputMethodWrapper.java
+++ b/core/java/android/inputmethodservice/IInputMethodWrapper.java
@@ -286,6 +286,10 @@ public void createSession(InputChannel channel, IInputSessionCallback callback)
 
     @Override
     public void setSessionEnabled(IInputMethodSession session, boolean enabled) {
+        if (session == null) {
+            Log.w(TAG, "Incoming session is null");
+            return;
+        }
         try {
             InputMethodSession ls = ((IInputMethodSessionWrapper)
                     session).getInternalInputMethodSession();
