Subject: [PATCH] Forward Port: Add Camera sound toggle [1/3]

diff --git a/media/java/android/media/MediaActionSound.java b/media/java/android/media/MediaActionSound.java
index 983ca754acd..ddfb24b5fc6 100644
--- a/media/java/android/media/MediaActionSound.java
+++ b/media/java/android/media/MediaActionSound.java
@@ -18,6 +18,7 @@
 
 import android.media.AudioManager;
 import android.media.SoundPool;
+import android.os.SystemProperties;
 import android.util.Log;
 
 /**
@@ -98,6 +99,7 @@
     private static final int STATE_LOADING                = 1;
     private static final int STATE_LOADING_PLAY_REQUESTED = 2;
     private static final int STATE_LOADED                 = 3;
+    private static final String PROP_CAMERA_SOUND = "persist.sys.camera-sound";
 
     private class SoundState {
         public final int name;
@@ -221,7 +223,9 @@ public void play(int soundName) {
                 sound.state = STATE_LOADING_PLAY_REQUESTED;
                 break;
             case STATE_LOADED:
-                mSoundPool.play(sound.id, 1.0f, 1.0f, 0, 0, 1.0f);
+                if (SystemProperties.getBoolean(PROP_CAMERA_SOUND, true)) {
+                    mSoundPool.play(sound.id, 1.0f, 1.0f, 0, 0, 1.0f);
+                }
                 break;
             default:
                 Log.e(TAG, "play() called in wrong state: " + sound.state + " for sound: "+ soundName);
