Subject: [PATCH] Fix double free issue in ResTable.

diff --git a/libs/androidfw/ResourceTypes.cpp b/libs/androidfw/ResourceTypes.cpp
index 38d2a58..55078eb 100644
--- a/libs/androidfw/ResourceTypes.cpp
+++ b/libs/androidfw/ResourceTypes.cpp
@@ -3263,6 +3263,7 @@
                 cacheEntry.filteredConfigs.clear();
 
                 bag_set** typeBags = cacheEntry.cachedBags;
+                cacheEntry.cachedBags = NULL;
                 if (kDebugTableNoisy) {
                     printf("typeBags=%p\n", typeBags);
                 }
@@ -3275,10 +3276,10 @@
                     for (size_t j = 0; j < N; j++) {
                         if (typeBags[j] && typeBags[j] != (bag_set*)0xFFFFFFFF) {
                             free(typeBags[j]);
+                            typeBags[j] = NULL;
                         }
                     }
                     free(typeBags);
-                    cacheEntry.cachedBags = NULL;
                 }
             }
         }
