Subject: [PATCH] Screen off Animations [2/2]

diff --git a/res/values/cr_arrays.xml b/res/values/cr_arrays.xml
index b2585d217c3..4aa3dac0af4 100644
--- a/res/values/cr_arrays.xml
+++ b/res/values/cr_arrays.xml
@@ -29,4 +29,17 @@
         <item>2</item>
     </string-array>
 
+    <!-- Screen off animation -->
+    <string-array name="screen_off_animation_entries">
+        <item>@string/screen_off_animation_default</item>
+        <item>@string/screen_off_animation_crt</item>
+        <item>@string/screen_off_animation_scale</item>
+    </string-array>
+
+    <string-array name="screen_off_animation_values" translatable="false">
+        <item>0</item>
+        <item>1</item>
+        <item>2</item>
+    </string-array>
+
 </resources>
diff --git a/res/values/cr_strings.xml b/res/values/cr_strings.xml
index 085b274ca25..92c023a7de7 100644
--- a/res/values/cr_strings.xml
+++ b/res/values/cr_strings.xml
@@ -72,4 +72,10 @@
     <string name="battery_stats_message">Battery history stats are going to be reset</string>
     <string name="ok_string">Ok</string>
 
+    <!-- Screen Off animation -->
+    <string name="screen_off_animation_title">Screen off animation</string>
+    <string name="screen_off_animation_default">Default</string>
+    <string name="screen_off_animation_crt">CRT</string>
+    <string name="screen_off_animation_scale">Scale</string>
+
 </resources>
diff --git a/res/xml/animation_settings.xml b/res/xml/animation_settings.xml
index e210fc6e..165a9ca9 100644
--- a/res/xml/animation_settings.xml
+++ b/res/xml/animation_settings.xml
@@ -24,4 +24,11 @@
         android:entries="@array/entries_scrollingcache"
         android:entryValues="@array/values_scrollingcache" />
 
+    <ListPreference
+        android:key="screen_off_animation"
+        android:title="@string/screen_off_animation_title"
+        android:entries="@array/screen_off_animation_entries"
+        android:entryValues="@array/screen_off_animation_values"
+        android:persistent="false" />
+
 </PreferenceScreen>
diff --git a/src/com/android/settings/cyanogenmod/AnimationSettings.java b/src/com/android/settings/cyanogenmod/AnimationSettings.java
index e00bd9f9..45e67d8c 100644
--- a/src/com/android/settings/cyanogenmod/AnimationSettings.java
+++ b/src/com/android/settings/cyanogenmod/AnimationSettings.java
@@ -48,10 +48,12 @@ public SettingsPreferenceFragment() {
     private static final String SCROLLINGCACHE_PREF = "pref_scrollingcache";
     private static final String SCROLLINGCACHE_PERSIST_PROP = "persist.sys.scrollingcache";
     private static final String SCROLLINGCACHE_DEFAULT = "1";
+    private static final String SCREEN_OFF_ANIMATION = "screen_off_animation";
 
     private Context mContext;
 
     private ListPreference mScrollingCachePref;
+    private ListPreference mScreenOffAnimations;
 
     @Override
     protected int getMetricsCategory() {
@@ -74,6 +76,14 @@ public void onCreate(Bundle savedInstanceState) {
         mScrollingCachePref.setValue(SystemProperties.get(SCROLLINGCACHE_PERSIST_PROP,
                 SystemProperties.get(SCROLLINGCACHE_PERSIST_PROP, SCROLLINGCACHE_DEFAULT)));
         mScrollingCachePref.setOnPreferenceChangeListener(this);
+
+        // Screen Off Animations
+        mScreenOffAnimations = (ListPreference) prefSet.findPreference(SCREEN_OFF_ANIMATION);
+        mScreenOffAnimations.setValue(String.valueOf(Settings.System.getInt(
+                resolver, Settings.System.SCREEN_OFF_ANIMATION, 0)));
+        mScreenOffAnimations.setSummary(mScreenOffAnimations.getEntry());
+        mScreenOffAnimations.setOnPreferenceChangeListener(this);
+
     }
 
     @Override
@@ -89,6 +99,12 @@ public boolean onPreferenceChange(Preference preference, Object newValue) {
                 SystemProperties.set(SCROLLINGCACHE_PERSIST_PROP, (String) newValue);
             }
             return true;
+        } else if (preference == mScreenOffAnimations) {
+            Settings.System.putInt(resolver, Settings.System.SCREEN_OFF_ANIMATION,
+                    Integer.parseInt((String) newValue));
+            mScreenOffAnimations.setValue(String.valueOf(newValue));
+            mScreenOffAnimations.setSummary(mScreenOffAnimations.getEntry());
+            return true;
         }
         return false;
     }
