Subject: [PATCH] Telecomm: Add Flashlight blink delay [3/3]

diff --git a/src/com/android/server/telecom/Ringer.java b/src/com/android/server/telecom/Ringer.java
index fcc57bc6fd50..845974a220c2 100644
--- a/src/com/android/server/telecom/Ringer.java
+++ b/src/com/android/server/telecom/Ringer.java
@@ -290,7 +290,7 @@
 
         private boolean shouldStop = false;
         private CameraManager cameraManager;
-        private int duration = 500;
+        private int mFlashlightDelay;
         private boolean hasFlash = true;
         private Context context;
 
@@ -310,15 +310,17 @@
 
         @Override
         protected Object doInBackground(Object[] objects) {
+            mFlashlightDelay = Settings.System.getInt(context.getContentResolver(),
+                Settings.System.FLASHLIGHT_DELAY, 500);
             if (hasFlash) {
                 try {
                     String cameraId = cameraManager.getCameraIdList()[0];
                     while (!shouldStop) {
                         cameraManager.setTorchMode(cameraId, true);
-                        Thread.sleep(duration);
+                        Thread.sleep(mFlashlightDelay);
 
                         cameraManager.setTorchMode(cameraId, false);
-                        Thread.sleep(duration);
+                        Thread.sleep(mFlashlightDelay);
                     }
                 } catch (Exception e) {
                     e.printStackTrace();
