FSubject: [PATCH] Fix null point exception in WriteSettingsDetails

diff --git a/src/com/android/settings/applications/WriteSettingsDetails.java b/src/com/android/settings/applications/WriteSettingsDetails.java
index d310d2b..a80cae4 100644
--- a/src/com/android/settings/applications/WriteSettingsDetails.java
+++ b/src/com/android/settings/applications/WriteSettingsDetails.java
@@ -134,6 +134,10 @@
 
     @Override
     protected boolean refreshUi() {
+        if (mPackageInfo.applicationInfo == null) {
+            return false;
+        }
+
         mWriteSettingsState = mAppBridge.getWriteSettingsInfo(mPackageName,
                 mPackageInfo.applicationInfo.uid);
 
