Subject: [PATCH] Disable/Enable screenshot sound [2/2]

diff --git a/res/values/cr_strings.xml b/res/values/cr_strings.xml
index a2851ea50a..81fc93251f 100644
--- a/res/values/cr_strings.xml
+++ b/res/values/cr_strings.xml
@@ -91,4 +91,7 @@
     <string name="camera_sounds_title">Camera shutter sound</string>
     <string name="camera_sounds_summary">Disabling may not work on 3rd party camera apps and in HDR+ mode</string>
 
+    <!-- Screenshot sound -->
+    <string name="screenshot_sound_title">Screenshot sound</string>
+
 </resources>
diff --git a/res/xml/other_sound_settings.xml b/res/xml/other_sound_settings.xml
index a0b053ea9f..1d05baec0d 100644
--- a/res/xml/other_sound_settings.xml
+++ b/res/xml/other_sound_settings.xml
@@ -25,6 +25,12 @@
             android:key="dial_pad_tones"
             android:title="@string/dial_pad_tones_title" />
 
+    <!-- Screenshot sounds -->
+    <SwitchPreference
+            android:key="screenshot_sound"
+            android:title="@string/screenshot_sound_title"
+            android:defaultValue="true" />
+
     <!-- Screen locking sounds -->
     <SwitchPreference
             android:key="screen_locking_sounds"
diff --git a/src/com/android/settings/notification/OtherSoundSettings.java b/src/com/android/settings/notification/OtherSoundSettings.java
index a5c36a359a..44f91a6efe 100644
--- a/src/com/android/settings/notification/OtherSoundSettings.java
+++ b/src/com/android/settings/notification/OtherSoundSettings.java
@@ -70,6 +70,7 @@
     private static final String KEY_SCREEN_LOCKING_SOUNDS = "screen_locking_sounds";
     private static final String KEY_CHARGING_SOUNDS = "charging_sounds";
     private static final String KEY_DOCKING_SOUNDS = "docking_sounds";
+    private static final String KEY_SCREENSHOT_SOUND = "screenshot_sound";
     private static final String KEY_TOUCH_SOUNDS = "touch_sounds";
     private static final String KEY_VIBRATE_ON_TOUCH = "vibrate_on_touch";
     private static final String KEY_DOCK_AUDIO_MEDIA = "dock_audio_media";
@@ -103,6 +104,11 @@ public boolean isApplicable(Context context) {
         }
     };
 
+    private static final SettingPref PREF_SCREENSHOT_SOUND = new SettingPref(
+            TYPE_SYSTEM, KEY_SCREENSHOT_SOUND, System.SCREENSHOT_SOUND,
+            DEFAULT_ON) {
+    };
+
     private static final SettingPref PREF_TOUCH_SOUNDS = new SettingPref(
             TYPE_SYSTEM, KEY_TOUCH_SOUNDS, System.SOUND_EFFECTS_ENABLED, DEFAULT_ON) {
         @Override
@@ -181,6 +187,7 @@ protected String getCaption(Resources res, int value) {
         PREF_SCREEN_LOCKING_SOUNDS,
         PREF_CHARGING_SOUNDS,
         PREF_DOCKING_SOUNDS,
+        PREF_SCREENSHOT_SOUND,
         PREF_TOUCH_SOUNDS,
         PREF_VIBRATE_ON_TOUCH,
         PREF_DOCK_AUDIO_MEDIA,
