Subject: [PATCH] Option to Display Data Disabled indicator Icons [2/2]

diff --git a/res/values/cr_strings.xml b/res/values/cr_strings.xml
index 8c2f943..1a95245 100644
--- a/res/values/cr_strings.xml
+++ b/res/values/cr_strings.xml
@@ -41,4 +41,8 @@
     <string name="qs_rows_landscape_title">Rows (landscape)</string>
     <string name="qs_columns_title">Columns</string>
 
+    <!-- Data disabled icon -->
+    <string name="data_disabled_icon_title">Show data disabled icon</string>
+    <string name="data_disabled_icon_summary">Display the cross sign next to signal bar when data is disabled</string>
+
 </resources>
diff --git a/res/xml/status_bar_settings.xml b/res/xml/status_bar_settings.xml
index 2d36cd4..38e0574 100644
--- a/res/xml/status_bar_settings.xml
+++ b/res/xml/status_bar_settings.xml
@@ -43,6 +43,12 @@
             android:summary="@string/status_bar_notif_count_summary"
             android:defaultValue="true" />
 
+        <cyanogenmod.preference.SystemSettingSwitchPreference
+            android:key="data_disabled_icon"
+            android:title="@string/data_disabled_icon_title"
+            android:summary="@string/data_disabled_icon_summary"
+            android:defaultValue="false" />
+
         <PreferenceScreen
             android:key="status_bar_icons"
             android:title="@string/status_bar_system_icons_title"
diff --git a/src/org/cyanogenmod/cmparts/statusbar/StatusBarSettings.java b/src/org/cyanogenmod/cmparts/statusbar/StatusBarSettings.java
index caf9d38..1269c21 100644
--- a/src/org/cyanogenmod/cmparts/statusbar/StatusBarSettings.java
+++ b/src/org/cyanogenmod/cmparts/statusbar/StatusBarSettings.java
@@ -19,6 +19,9 @@
 import android.os.Bundle;
 import android.support.v7.preference.Preference;
 import android.support.v7.preference.Preference.OnPreferenceChangeListener;
+import android.support.v14.preference.SwitchPreference;
+import android.telephony.TelephonyManager;
+import android.telephony.SubscriptionManager;
 import android.text.format.DateFormat;
 import android.view.View;
 
@@ -35,6 +38,11 @@
     private static final String STATUS_BAR_BATTERY_STYLE = "status_bar_battery_style";
     private static final String STATUS_BAR_SHOW_BATTERY_PERCENT = "status_bar_show_battery_percent";
     private static final String STATUS_BAR_QUICK_QS_PULLDOWN = "qs_quick_pulldown";
+    private static final String DATA_DISABLED_PREF = "data_disabled_icon";
+
+    private SubscriptionManager mSm;
+
+    private SwitchPreference mNoData;
 
     private static final int STATUS_BAR_BATTERY_STYLE_HIDDEN = 4;
     private static final int STATUS_BAR_BATTERY_STYLE_TEXT = 6;
@@ -72,6 +80,13 @@ public void onCreate(Bundle savedInstanceState) {
                 (CMSystemSettingListPreference) findPreference(STATUS_BAR_QUICK_QS_PULLDOWN);
         mQuickPulldown.setOnPreferenceChangeListener(this);
         updateQuickPulldownSummary(mQuickPulldown.getIntValue(0));
+
+        mNoData = (SwitchPreference) findPreference(DATA_DISABLED_PREF);
+        mSm = (SubscriptionManager) getSystemService(getContext().TELEPHONY_SUBSCRIPTION_SERVICE);
+
+        if (mSm.getActiveSubscriptionInfoCount() <= 0) {
+                getPreferenceScreen().removePreference(mNoData);
+        }
     }
 
     @Override
