Subject: [PATCH] Avoid telecom crash

mRingingCalls may not have call when startCallWaiting be invoked

diff --git a/src/com/android/server/telecom/CallAudioManager.java b/src/com/android/server/telecom/CallAudioManager.java
index 227d5f5..6a91b18 100644
--- a/src/com/android/server/telecom/CallAudioManager.java
+++ b/src/com/android/server/telecom/CallAudioManager.java
@@ -433,7 +433,9 @@
 
     @VisibleForTesting
     public void startCallWaiting() {
-        mRinger.startCallWaiting(mRingingCalls.iterator().next());
+        if (mRingingCalls.size() == 1) {
+            mRinger.startCallWaiting(mRingingCalls.iterator().next());
+        }
     }
 
     @VisibleForTesting
