Subject: [PATCH] PublicVolume: don't override mFsLabel

diff --git a/PublicVolume.cpp b/PublicVolume.cpp
index 8faf00d..f6caf63 100644
--- a/PublicVolume.cpp
+++ b/PublicVolume.cpp
@@ -62,10 +62,10 @@ PublicVolume::~PublicVolume() {
 }
 
 status_t PublicVolume::readMetadata() {
-    status_t res = ReadMetadataUntrusted(mDevPath, mFsType, mFsUuid, mFsLabel);
+    std::string unused;
+    status_t res = ReadMetadataUntrusted(mDevPath, mFsType, mFsUuid, unused);
     notifyEvent(ResponseCode::VolumeFsTypeChanged, mFsType);
     notifyEvent(ResponseCode::VolumeFsUuidChanged, mFsUuid);
-    notifyEvent(ResponseCode::VolumeFsLabelChanged, mFsLabel);
     return res;
 }
 
