Don't start tethering if IPv6 RPF is not supported. 

diff --git a/server/NatController.cpp b/server/NatController.cpp
index cbc3611..4129c0f 100644
--- a/server/NatController.cpp
+++ b/server/NatController.cpp
@@ -376,7 +376,8 @@ int NatController::setForwardRules(bool add, const char *intIface, const char *e
         goto err_return;
     }
 
-    if (runCmd(ARRAY_SIZE(cmd4), cmd4) && add) {
+    // STOPSHIP: Make this an error.
+    if (runCmd(ARRAY_SIZE(cmd4), cmd4) && add && false /* STOPSHIP */) {
         rc = -1;
         goto err_rpfilter;
     }
