Subject: [PATCH 43/54] Revert "Add a protected broadcast for Masquerade events"

diff --git a/core/res/AndroidManifest.xml b/core/res/AndroidManifest.xml
index 6af7d84f53d..5335ca41259 100644
--- a/core/res/AndroidManifest.xml
+++ b/core/res/AndroidManifest.xml
@@ -530,8 +530,6 @@
     <!-- ====================================================================== -->
     <eat-comment />
 
-    <protected-broadcast android:name="masquerade.substratum.STATUS_CHANGED" />
-
     <!-- ====================================================================== -->
     <!-- Permissions for accessing user's contacts including personal profile   -->
     <!-- ====================================================================== -->
diff --git a/services/core/java/com/android/server/pm/PackageManagerService.java b/services/core/java/com/android/server/pm/PackageManagerService.java
index 904bc5bd548..33bbf257f1c 100644
--- a/services/core/java/com/android/server/pm/PackageManagerService.java
+++ b/services/core/java/com/android/server/pm/PackageManagerService.java
@@ -4584,8 +4584,7 @@ public class PackageManagerService extends IPackageManager.Stub {
                 if (actionName.startsWith("android.net.netmon.lingerExpired")
                         || actionName.startsWith("com.android.server.sip.SipWakeupTimer")
                         || actionName.startsWith("com.android.internal.telephony.data-reconnect")
-                        || actionName.startsWith("android.net.netmon.launchCaptivePortalApp")
-                        || actionName.startsWith("masquerade.substratum.STATUS_CHANGED")) {
+                        || actionName.startsWith("android.net.netmon.launchCaptivePortalApp")) {
                     return true;
                 }
             }
