Subject: [PATCH] sepolicy: Add Color Engine policy

diff --git a/sepolicy/service.te b/sepolicy/service.te
index c7ad50f..ac1f8e2 100644
--- a/sepolicy/service.te
+++ b/sepolicy/service.te
@@ -9,3 +9,4 @@
 type cm_weather_service, system_api_service, system_server_service, service_manager_type;
 type cm_livedisplay_service, system_api_service, system_server_service, service_manager_type;
 type cm_audio_service, system_api_service, system_server_service, service_manager_type;
+type theme_service, system_api_service, system_server_service, service_manager_type;
diff --git a/sepolicy/service_contexts b/sepolicy/service_contexts
index 90f21c9..a5e3e50 100644
--- a/sepolicy/service_contexts
+++ b/sepolicy/service_contexts
@@ -9,3 +9,4 @@
 cmweather                                 u:object_r:cm_weather_service:s0
 cmlivedisplay                             u:object_r:cm_livedisplay_service:s0               
 cmaudio                                   u:object_r:cm_audio_service:s0
+theme                                     u:object_r:theme_service:s0
diff --git a/sepolicy/system_server.te b/sepolicy/system_server.te
index 61b8565..e95c24c 100644
--- a/sepolicy/system_server.te
+++ b/sepolicy/system_server.te
@@ -1,6 +1,7 @@
 allow system_server cache_recovery_file:dir rw_dir_perms;
 allow system_server cache_recovery_file:file create_file_perms;
 allow system_server cache_recovery_file:fifo_file create_file_perms;
+allow system_server theme_service:service_manager { add find };
 
 # Persistent properties
 allow system_server persist_property_file:dir rw_dir_perms;
diff --git a/sepolicy/untrusted_app.te b/sepolicy/untrusted_app.te
index 2372f16..1c814ce 100644
--- a/sepolicy/untrusted_app.te
+++ b/sepolicy/untrusted_app.te
@@ -1,3 +1,4 @@
 allow untrusted_app cm_weather_service:service_manager find;
 allow untrusted_app cm_status_bar_service:service_manager find;
 allow untrusted_app cm_profile_service:service_manager find;
+allow untrusted_app theme_service:service_manager find;
